### PR TITLE
Add io

### PR DESCRIPTION
**Pull request by Armin12**

### CODE REVIEW

File: src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+def read_transcription_data(path_to_files, filename_pattern):
+    fpath0 = os.path.join(path_to_files, filename_pattern)
+    f_lngg_level = glob.glob(fpath0)[0]
+    with open(f_lngg_level, "r") as f:
+        lngg_level = []
+        for x in f:
+            lngg_level.append(x.split())
+    f.close()
```
Comment (bendichter) [inline src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py]:
```suggestion
```

File: src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# phonemes
+for phonemes_sample in phonemes_data:
+    phonemes.add_interval(label=phonemes_sample[2], start_time=float(phonemes_sample[0]), stop_time=float(phonemes_sample[1]))
+
+# syllables
+## To figure out how to create syllables_keys automatically?
+syllables_keys = {'0': 'h#', '1': 'bricks', '2': 'are', '3': 'an', '4': 'al', 
+                  '5': 'ter', '6': 'na', '7': 'tive'}
+
+syllables.add_column('phonemes', '')
```
Comment (bendichter) [inline src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py]: no empty strings. Put 'no description' if you really can't think of a description

File: src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+syllables.add_column('word_onset', '')
+syllables.add_column('syllable_number', '')
+syllables.add_column('speech_sound', '')
+
+
+for j, syllables_sample in enumerate(syllables_data):
+    syllables.add_interval(label=syllables_keys[syllables_sample[4]],
+                           start_time=float(syllables_sample[0]),
+                           stop_time=float(syllables_sample[1]),
+                           phonemes=syllables_sample[2],
```
Comment (bendichter) [inline src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py]: phonemes should be in their own table

File: src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+# syllables
+## To figure out how to create syllables_keys automatically?
+syllables_keys = {'0': 'h#', '1': 'bricks', '2': 'are', '3': 'an', '4': 'al', 
+                  '5': 'ter', '6': 'na', '7': 'tive'}
+
+syllables.add_column('phonemes', '')
+syllables.add_column('word_onset', '')
+syllables.add_column('syllable_number', '')
+syllables.add_column('speech_sound', '')
```
Comment (bendichter) [inline src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py]: should this be "label"?

File: src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    sentences_data[i] = sentences_data[i][0:2] + [' '.join(sentences_data[i][2:])]
+
+
+# phonemes
+for phonemes_sample in phonemes_data:
+    phonemes.add_interval(label=phonemes_sample[2], start_time=float(phonemes_sample[0]), stop_time=float(phonemes_sample[1]))
+
+# syllables
+## To figure out how to create syllables_keys automatically?
+syllables_keys = {'0': 'h#', '1': 'bricks', '2': 'are', '3': 'an', '4': 'al',
```
Comment (bendichter) [inline src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py]: we need to figure out how to automate this

File: src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
+
+import os
+import glob
+from ndx_hierarchical_behavioral_data.definitions.transcription import phonemes, syllables, words, sentences
+
+
+def read_transcription_data(path_to_files, filename_pattern):
```
Comment (bendichter) [inline src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py]: pandas.read_csv might be better here.

**Review by bendichter — CHANGES_REQUESTED** [review]

File: src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py (hunk 2 above)
Comment (Armin12) [reply src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py]: Comments were added.

File: src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py (hunk 3 above)
Comment (Armin12) [reply src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py]: Yes, phonemes were removed from syllables.

File: src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py (hunk 4 above)
Comment (Armin12) [reply src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py]: syllable_number are indices of labels. So, this part was changed accordingly.

File: src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py (hunk 6 above)
Comment (Armin12) [reply src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py]: Data files are not in .csv format and also do not have headings.

Comment (bendichter) [reply src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py]:
```python
import pandas as pd
pd.read_csv('mbbr0_si2315.phn',
            names=['start_time', 'stop_time', 'label'],
            sep=' ')
```

File: src/pynwb/ndx_hierarchical_behavioral_data/mocha_io.py [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+                trial_id='...'):
+    """Given phonemes, syllables, words, and sentences data, this function extracts specified id's
+
+        For given phonemes, syllables, words, and sentences data, this function extracts information about
+        specified trials/session/or subjects
+
+        Parameters
+        ----------
+        phoneme_data: pandas.DataFrame
+            phonemes' DataFrame
```
Comment (bendichter) [inline src/pynwb/ndx_hierarchical_behavioral_data/mocha_io.py]: what is the purpose of these apostrophes?

File: src/pynwb/ndx_hierarchical_behavioral_data/text_grid_io.py [hunk 8]
```diff
@@ -0,0 +1,100 @@
+import os
+import glob
+import pandas as pd
+import re
+from pynwb.epoch import TimeIntervals
+
+
+def textgriddf_reader(path_to_files, filename_pattern='*TextGrid'):
+    """Read TextGrid file and format it by \n
```
Comment (bendichter) [inline src/pynwb/ndx_hierarchical_behavioral_data/text_grid_io.py]: Don't use \n

File: src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py [hunk 9]
```diff
@@ -0,0 +1,141 @@
+import os
+import glob
+import pandas as pd
+import numpy as np
+import re
+from pynwb import TimeSeries
+from ndx_hierarchical_behavioral_data.definitions.transcription import phonemes, syllables, words, sentences
+
+
+def timitsounds_reader(path_to_files, filename_pattern, add_headings, separator=' '):
```
Comment (bendichter) [inline src/pynwb/ndx_hierarchical_behavioral_data/transcription_io.py:10]: please add docstrings in this file as well

**Review by bendichter — CHANGES_REQUESTED** [review]

**Pull request merged**